samples/apps/copilot-chat-app/webapp/src/App.tsx

```diff
@@ -69,26 +70,38 @@ const App: FC = () => {
     const dispatch = useAppDispatch();
 
     const { instance, inProgress } = useMsal();
-    const account = instance.getActiveAccount();
+    const { loggedInUserInfo } = useAppSelector((state: RootState) => state.conversations);
     const isAuthenticated = useIsAuthenticated();
 
     const chat = useChat();
 
     useEffect(() => {
-        if (isAuthenticated && account) {
-            dispatch(setLoggedInUserId(account.homeAccountId));
-
-            if (appState === AppState.LoadingChats) {
-                // Load all chats from the backend.
-                async function loadChats() {
-                    if (await chat.loadChats()) {
-                        setAppState(AppState.Chat);
-                    }
+        if (isAuthenticated) {
+            if (loggedInUserInfo === undefined) {
```

Comment:
Lets add a timestamp to the loggedInUserInfo object. If that timestamp is more than x mins old, say 30 mins, then we also try and fetch the activeAccount.

Reply:
Why do we need to fetch the account information if it hasn't changed? It will only change if the user signs out.

Reply:
The thought was more from a security perspective and logging out stale accounts. In retrospect its probably something someone adapting the sample app should look at based on their use case.